### ISSUE TITLE
Accessibility -DatePicker - focus should stay inside callout until date is selected or picker is dismissed

### ISSUE THREAD

**Issue by daweffe** [OfficeDev/office-ui-fabric-react#4415]
### Bug Report

- __Package version(s)__: 5.57
- __Browser and OS versions__: all

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/sites affected: Planner

#### Describe the issue:

When focus is inside the date picker component, you are able to tab outside of the datepicker. Focus needs to stay inside the callout until a date is picked or the picker is dismissed with a click outside the callout or the ESC key. We have an accessibility bug filed against Planner for this issue.

**Comment by Jahnp** [comment]
Thanks for reporting this. @lorejoh12, @sathi23 this affects DatePicker--can you or someone from your team take a look at this?

**Comment by daweffe** [comment]
I have also noticed that trying to place content inside a FocusTrapZone does not work right when that content contains a FocusZone. You may have to fix that issue to fix this, and I would like that fix as well.

**Comment by lorejoh12** [comment]
adding this issue to the triage list for next week, will update with expected timeline

**Comment by lorejoh12** [comment]
We're working on a fix for this right now, will hopefully have the fix in either this week or next week. Most popups in Fabric right now are wrapped in FocusTrapZones, so that when tabbing to the end of the popup the focus can cycle back around. That seems like a reasonable solution here also.

From initial testing, it seems to work okay, although there is an issue with both the FocusTrapZone and the DatePicker trying to reset focus after closing the control. This is related to #4623 so hopefully the fix here fixes both issues.
